doc/decisions/boolean.md

```diff
@@ -6,46 +6,47 @@ Inconsistent use of bool in various parts of Elektra.
 
 ## Constraints
 
+- needs to be string
+
 ## Assumptions
 
-- needs to be string
-- convenience plugins can convert anything to 0 or 1
 - type checker plugins can reject everything not 0 or 1
 
 ## Considered Alternatives
 
-- strictly only allow 0 and 1 (would move validation across the code)
 - only check presence or absence (no cascading override of already present key possible)
-- use as in CMake (would move convenience across the code)
+- use booleans as in CMake, which allows on/off, true/false, ... (would need convenience across the code)
 
 ## Decision
 
-Use, depending on what your default should be:
+Only the strings `0` and `1` are allowed in the `KeySet` for `type = boolean`.
+Everything else should lead to errors in checkers (in `kdbSet`).
+
+Storage plugins are allowed any representation as suitable.
 
-- 0 is false, everything else is true (default is true)
-- 1 is true, everything else is false (default is false)
+In the absence of the key, the default can be either:
 
-Example:
+- default is true:
+  `0` is false, everything else is true, or
+- default is false:
+  `1` is true, everything else is false.
```

Comment:
Yes, accepting only 0 and 1 is totally legitimate.

Only applications being too lax is problematic, as it leads to undetected misconfiguration (e.g. the admin thinks it will be true but the application actually interpreted it as false). I made it even more strict: 9b40823114cbcc91ba73ad9efef103427c229ce3

Reply:
> The spec/docu should mention that a `boolean` is used and may specify the default.
> The absence of a non-required key without a specified default should be interpreted as false.

I think this (plus your C code) is a rather convoluted way of saying:

Only keys defined as `boolean` by the spec or by documentation should be treated as boolean. If no default value is specified the key should be interpreted as false. Users should assume that `1` is the only value that is interpreted as true.

We don't need to bring the notion of required keys into the mix here, since those are handled elsewhere and are totally independent of the type system.

Reply:
Thank you, I updated the text: e5394e5d5